src/bid_order.rs

```diff
@@ -162,197 +135,21 @@ impl BidOrderV2 {
-    env: Env,
+    _env: Env,
     _msg: &MigrateMsg,
-    mut response: Response<ProvenanceMsg>,
+    response: Response<ProvenanceMsg>,
 ) -> Result<Response<ProvenanceMsg>, ContractError> {
     let store = deps.storage;
-    let querier = deps.querier;
     let version_info = get_version_info(store)?;
     let current_version = Version::parse(&version_info.version)?;
 
-    // version support added in 0.15.0, all previous versions migrate to v1 of state data
-    let upgrade_req = VersionReq::parse("<0.15.0")?;
-
-    if upgrade_req.matches(&current_version) {
-        let existing_bid_order_ids: Vec<Vec<u8>> = bucket_read(store, NAMESPACE_ORDER_BID)
-            .range(None, None, Order::Ascending)
-            .map(|kv_bid: StdResult<Record<BidOrder>>| {
-                let (bid_key, _) = kv_bid.unwrap();
-                bid_key
-            })
-            .collect();
-
-        for existing_bid_order_id in existing_bid_order_ids {
-            let existing_bid_order: BidOrder =
-                bucket_read(store, NAMESPACE_ORDER_BID).load(&existing_bid_order_id)?;
-            get_bid_storage(store).save(&existing_bid_order_id, &existing_bid_order.into())?
-        }
-    }
-
-    // migration from 0.15.2 - 0.16.1 => latest
-    if VersionReq::parse(">=0.15.1, <0.16.2")?.matches(&current_version) {
-        let existing_bid_order_ids: Vec<Vec<u8>> = bucket_read(store, NAMESPACE_ORDER_BID)
-            .range(None, None, Order::Ascending)
-            .map(|kv_bid: StdResult<Record<BidOrderV1>>| {
-                let (bid_key, _) = kv_bid.unwrap();
-                bid_key
-            })
-            .collect();
-
-        for existing_bid_order_id in existing_bid_order_ids {
-            let existing_bid_order: BidOrderV1 =
-                bucket_read(store, NAMESPACE_ORDER_BID).load(&existing_bid_order_id)?;
-            get_bid_storage(store).save(&existing_bid_order_id, &existing_bid_order.into())?
-        }
-    }
-
-    // bid fees and order events added in 0.16.3, refunds may be necessary for existing orders
-    if VersionReq::parse("<0.16.3")?.matches(&current_version) {
-        // get all bid ids
-        let existing_bid_order_ids: Vec<Vec<u8>> = get_bid_storage_read(store)
-            .range(None, None, Order::Ascending)
-            .map(|kv_bid| {
-                let (bid_key, _) = kv_bid.unwrap();
-                bid_key
-            })
-            .collect();
-
-        // determine and create a refund if necessary
-        for existing_bid_order_id in existing_bid_order_ids {
-            let mut existing_bid_order =
-                get_bid_storage_read(store).load(&existing_bid_order_id)?;
-
-            response = calculate_migrate_refund(&querier, &env, response, &mut existing_bid_order)
-                .unwrap();
-
-            get_bid_storage(store).save(&existing_bid_order_id, &existing_bid_order)?
-        }
-    }
+    require_version(">=0.16.2", &current_version)?;
```

Comment:
maybe same to check the old version migration fails, but could also just use the `migrate()` entry point to check that